winml/test/common/taefTestMacros.h

```diff
@@ -48,14 +50,17 @@ using namespace WEX::TestExecution;
 
 #ifndef USE_DML
 #define GPUTEST \
-  WINML_SUPRESS_UNREACHABLE_BELOW(WINML_SKIP_TEST("GPU tests disabled because this is a WinML only build (no DML)"))
+  WINML_SKIP_TEST("GPU tests disabled because this is a WinML only build (no DML)")
+#define GPUTEST_ENABLED alwaysFalse()
 #else
 #define GPUTEST                                                                             \
   bool noGPUTests;                                                                          \
   if (SUCCEEDED(RuntimeParameters::TryGetValue(L"noGPUtests", noGPUTests)) && noGPUTests) { \
     WINML_SKIP_TEST("This test is disabled by the noGPUTests runtime parameter.");          \
     return;                                                                                 \
   }
+#define GPUTEST_ENABLED bool noGPUTests; \
```

Comment:
All of these macros should really be defined with:

#define foo do { ... } while(0)

In this case it will add a new scope and prevents weird errors in the case that someone accidentally defines noGPUTests. Also, it will force a compile time error if a callers doesn't add a semicolon at the end of the macro and ensure consistency, rather than some people adding it sometimes, and forgetting other times.

Reply:
It's meant to be used in a if condition: `if (GPUTEST_ENABLED)` , so the scope is limited.